api/src/database/migrations/20220708A-update-notifications-timestamp-column.ts

```diff
@@ -0,0 +1,13 @@
+import { Knex } from 'knex';
+
+export async function up(knex: Knex): Promise<void> {
+	await knex.schema.alterTable('directus_notifications', (table) => {
+		table.timestamp('timestamp').defaultTo(knex.fn.now()).alter();
+	});
+}
+
+export async function down(knex: Knex): Promise<void> {
+	await knex.schema.alterTable('directus_notifications', (table) => {
+		table.timestamp('timestamp').notNullable().alter();
```

Comment:
Should this remove the default value again?

Reply:
Also we should use the helper like in other migrations. I'm pretty sure this breaks in oracle

Reply:
> Should this remove the default value again?

I believe not adding `defaultTo` in an alter statement means there won't be default value anymore 🤔 at least that's the case when I tested on Postgres, and I think this is also the same logic in the helper as well:

https://github.com/directus/directus/blob/06a618c758d0a7f45536b2c5e8eb9d27ccd6426b/api/src/database/helpers/schema/types.ts#L22-L44

> Also we should use the helper like in other migrations. I'm pretty sure this breaks in oracle

Actually I tried to find it before the current migration, but I'm not sure if there's one that's applicable for timestamp? (other than `changeToText`, `changeToInteger` and `changeToString`. It also seems like `changeNullable` isn't really used anywhere except in the protected `changeToTypeByCopy`)

Reply:
> Actually I tried to find it before the current migration, but I'm not sure if there's one that's applicable for timestamp? (other than changeToText, changeToInteger and changeToString

Hmm, we should just make a `updateType` helper with a string for the type input.

Reply:
added the `updateType` helper, but not sure am I misinterpreting the expected usage 🤔

but assuming it fits, this would mean `changeToText` and `changeToInteger` will technically be redundant right? as we can use `updateType(table, column, 'integer', options)` to replace `changeToInteger(table, column, options)` for example.

Reply:
> but assuming it fits, this would mean changeToText and changeToInteger will technically be redundant right? 

That's correct. I'd rather have one "changeType" utility that takes the types as parameters, than individual methods for each type 👍🏻

Reply:
Thanks for clearing it up 👍 replaced all the individual methods now.